### PR TITLE
log-backup: check timezone when using datatime format ts string

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/31975

Problem Summary:
In log backup. User may set wrong datatime of restore ts. This PR tried to add a timezone check when User not set tz in ts string.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by YuJuncen — APPROVED** [review]

File: br/pkg/task/backup.go [hunk 1]
```diff
@@ -137,7 +137,7 @@ func (cfg *BackupConfig) ParseFromFlags(flags *pflag.FlagSet) error {
 	if err != nil {
 		return errors.Trace(err)
 	}
-	cfg.BackupTS, err = ParseTSString(backupTS)
+	cfg.BackupTS, err = ParseTSString(backupTS, false)
```
Comment (joccau) [inline br/pkg/task/backup.go:140]: Why does backup full-snapshot use false?

**Review by joccau — COMMENTED** [review]
rest LGTM

File: br/pkg/task/backup.go (hunk 1 above)
Comment (3pointer) [reply br/pkg/task/backup.go:140]: Because We don't want to make breaking change here.

**Pull request merged**